extensions/libarchive/MergeContent.cpp

```diff
@@ -271,6 +281,16 @@ bool MergeContent::processBin(core::ProcessContext *context, core::ProcessSessio
       return false;
     }
     session->putAttribute(mergeFlow, BinFiles::FRAGMENT_COUNT_ATTRIBUTE, std::to_string(bin->getSize()));
+
+    std::unique_ptr<AttributeMerger> attributeMerger;
+    if (attributeStrategy_ == ATTRIBUTE_STRATEGY_KEEP_COMMON)
+      attributeMerger = std::unique_ptr < AttributeMerger > (new KeepOnlyCommonAttributesMerger(session, bin->getFlowFile(), mergeFlow));
```

Comment:
I would keep it on the stack and immediately use it like `KeepOnlyCommonAttributesMerger(session, bin->getFlowFile(), mergeFlow).mergeAttributes()` it really is a single-use object (and we have to be careful to keep it that way as `KeepAllUniqueAttributesMerger::mergeAttributes` is not idempotent)

Reply:
I see now that `mergeBin` follows the same pattern, although there `->merge` could throw